src/elements/content-explorer/ContentExplorer.js

```diff
@@ -780,6 +782,21 @@ class ContentExplorer extends Component<Props, State> {
         this.setState({ currentCollection: newCollection, selected: newSelectedItem }, callback);
     }
 
+    /**
+     * Update the item in state with the given id to have the given thumbnailUrl
+     *
+     * @param {string} id - id of BoxItem to update
+     * @param {string} thumbnailUrl - thumbnailUrl to update BoxItem with
+     * @return {void}
+     */
+    updateItemThumbnailUrl = (id: string, thumbnailUrl: string): void => {
+        const { currentCollection } = this.state;
+        const { items = [] } = currentCollection;
+        const newCollection = { ...currentCollection };
+        newCollection.items = items.map(item => (item.id === id ? { ...cloneDeep(item), thumbnailUrl } : item));
+        this.setState({ currentCollection: newCollection });
```

Comment:
This function will be called for each thumbnail that needs to be generated, which also means that `setState` will be called more often.  I think this may be worth it, since generating the thumbnails could vary between files, and it's better to display them one by one than wait for all of them to finish generating.